Failing test: Security Solution Cypress.x-pack/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button·cy·ts - Inspect Explore pages inspect Hosts page inspect Hosts page

A test failed on a tracked branch

```
AssertionError: Timed out retrying after 150000ms: Expected to find element: `.euiSuperSelect__item [data-test-subj="dataView-option-super"]`, but never found it.
    at selectDataView (webpack:///./tasks/sourcerer.ts:27:5)
    at Object.onLoad (webpack:///./e2e/explore/inspect/inspect_button.cy.ts:54:27)
    at Context.onLoad (webpack:///./tasks/navigation.ts:62:6)
```

First failure: [CI Build - main](https://buildkite.com/elastic/kibana-on-merge/builds/42360#018e2bf3-4bc3-4d36-94eb-e7add0a94585)

<!-- kibanaCiData = {"failed-test":{"test.class":"Security Solution Cypress.x-pack/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button·cy·ts","test.name":"Inspect Explore pages inspect Hosts page inspect Hosts page","test.failCount":4}} -->

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
@@ -24,12 +24,7 @@ import { mockRiskEngineEnabled } from '../../../tasks/entity_analytics';
 
 const DATA_VIEW = 'auditbeat-*';
 
-// The Hosts and Users pages in this suite are still individually flaky/skipped for reasons
-// tracked by their own issues below. Their root cause has not been verified as fixed, so they
-// stay skipped here. Do not remove these without separately verifying and closing those issues.
-// FLAKY (Hosts page): https://github.com/elastic/kibana/issues/178367
-// FLAKY (Users page): https://github.com/elastic/kibana/issues/199583
-const SKIPPED_PAGES = ['Hosts', 'Users'];
+const SKIPPED_PAGES: string[] = [];
 
 describe('Inspect Explore pages', { tags: ['@ess', '@serverless'] }, () => {
   beforeEach(() => {

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
@@ -20,7 +20,7 @@ import { login } from '../../../tasks/login';
 import { visitWithTimeRange } from '../../../tasks/navigation';
 import { waitForWelcomePanelToBeLoaded } from '../../../tasks/common';
 import { postDataView } from '../../../tasks/api_calls/common';
-import { mockRiskEngineEnabled } from '../../../tasks/entity_analytics';
+import { mockEntityStoreRiskScores, mockRiskEngineEnabled } from '../../../tasks/entity_analytics';
 
 const DATA_VIEW = 'auditbeat-*';
 
@@ -32,6 +32,7 @@ describe('Inspect Explore pages', { tags: ['@ess', '@serverless'] }, () => {
     cy.task('esArchiverLoad', { archiveName: 'risk_scores_new' });
     login();
     mockRiskEngineEnabled();
+    mockEntityStoreRiskScores();
     // Create and select data view
     postDataView(DATA_VIEW);
   });

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/explore/inspect/inspect_button.cy.ts
@@ -20,7 +20,11 @@ import { login } from '../../../tasks/login';
 import { visitWithTimeRange } from '../../../tasks/navigation';
 import { waitForWelcomePanelToBeLoaded } from '../../../tasks/common';
 import { postDataView } from '../../../tasks/api_calls/common';
-import { mockEntityStoreRiskScores, mockRiskEngineEnabled } from '../../../tasks/entity_analytics';
+import {
+  mockEntityStoreRiskScores,
+  mockRiskEngineEnabled,
+  mockRiskEnginePrivileges,
+} from '../../../tasks/entity_analytics';
 
 const DATA_VIEW = 'auditbeat-*';
 
@@ -32,6 +36,7 @@ describe('Inspect Explore pages', { tags: ['@ess', '@serverless'] }, () => {
     cy.task('esArchiverLoad', { archiveName: 'risk_scores_new' });
     login();
     mockRiskEngineEnabled();
+    mockRiskEnginePrivileges();
     mockEntityStoreRiskScores();
     // Create and select data view
     postDataView(DATA_VIEW);

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
@@ -93,7 +93,7 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
       {
         title: 'Host risk',
         tab: RISK_DETAILS_NAV,
-        customIndexPattern: 'risk-score.risk-score-latest-default',
+        customIndexPattern: '.entities.v2.latest.security',
         id: HOST_BY_RISK_TABLE,
       },
     ],
@@ -295,7 +295,7 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
         title: 'User risk',
         tab: RISK_SCORE_TAB,
         id: RISK_SCORE_TAB_CONTENT,
-        customIndexPattern: 'risk-score.risk-score-latest-default',
+        customIndexPattern: '.entities.v2.latest.security',
       },
     ],
   },

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
@@ -77,6 +77,7 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
         title: 'All Hosts Table',
         tab: ALL_HOSTS_TAB,
         id: ALL_HOSTS_TABLE,
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'Uncommon processes Table',
@@ -274,6 +275,7 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
         title: 'Users Table',
         tab: ALL_USERS_TAB,
         id: ALL_USERS_TABLE,
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'Destination IPs Table',

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/screens/inspect.ts
@@ -103,12 +103,14 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
         panelSelector: HOSTS_VISUALIZATION,
         tab: ALL_HOSTS_TAB,
         embeddableId: 'hostsKpiHostsQuery-hosts-metric-embeddable',
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'all hosts',
         panelSelector: HOSTS_VISUALIZATION,
         tab: ALL_HOSTS_TAB,
         embeddableId: 'hostsKpiHostsQuery-area-embeddable',
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'Unique IPs',
@@ -232,12 +234,14 @@ export const INSPECT_BUTTONS_IN_SECURITY: InspectButtonMetadata[] = [
         panelSelector: USERS_VISUALIZATION,
         embeddableId: 'TotalUsersKpiQuery-users-metric-embeddable',
         tab: ALL_USERS_TAB,
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'Users',
         panelSelector: USERS_VISUALIZATION,
         embeddableId: 'TotalUsersKpiQuery-area-embeddable',
         tab: ALL_USERS_TAB,
+        customIndexPattern: '.entities.v2.latest.security',
       },
       {
         title: 'User authentications',

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -52,6 +52,14 @@ export const navigateToNextPage = () => {
   cy.get(ANOMALIES_TABLE_NEXT_PAGE_BUTTON).click();
 };
 
+/**
+ * The entity maintainers endpoint is called instead of RISK_ENGINE_STATUS_URL when
+ * the `entityAnalyticsEntityStoreV2` experimental feature flag is enabled (default: true
+ * on main). Both URLs must be mocked so the risk tables resolve out of loading state
+ * regardless of which code path is active.
+ */
+const ENTITY_MAINTAINERS_URL = '/internal/security/entity_store/entity_maintainers*';
+
 export const mockRiskEngineEnabled = () => {
   cy.intercept('GET', RISK_ENGINE_STATUS_URL, {
     statusCode: 200,
@@ -59,6 +67,26 @@ export const mockRiskEngineEnabled = () => {
       risk_engine_status: 'ENABLED',
     },
   }).as('riskEngineStatus');
+
+  cy.intercept('GET', ENTITY_MAINTAINERS_URL, {
+    statusCode: 200,
+    body: {
+      maintainers: [
+        {
+          id: 'risk-score',
+          taskStatus: 'started',
+          interval: '1h',
+          description: null,
+          nextRunAt: null,
+          minLicense: 'platinum',
+          customState: null,
+          runs: 1,
+          lastSuccessTimestamp: null,
+          lastErrorTimestamp: null,
+        },
+      ],
+    },
+  }).as('entityMaintainers');
 };
 
 export const mockRiskEnginePrivileges = () => {

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -64,7 +64,9 @@ const ENTITY_STORE_ENTITIES_URL = '/api/security/entity_store/entities*';
 const entityStoreDsl = (entityIndex: string) =>
   JSON.stringify({
     index: [entityIndex],
-    body: { query: { bool: { filter: [{ exists: { field: 'entity.risk.calculated_score_norm' } }] } } },
+    body: {
+      query: { bool: { filter: [{ exists: { field: 'entity.risk.calculated_score_norm' } }] } },
+    },
   });
 
 /**
@@ -86,15 +88,39 @@ export const mockEntityStoreRiskScores = () => {
     const hostRecord = {
       '@timestamp': '2025-01-01T00:00:00.000Z',
       'host.name': 'mock-host',
-      entity: { name: 'mock-host', type: 'node', risk: { calculated_score_norm: 70, calculated_level: 'High' } },
-      host: { name: 'mock-host', risk: { calculated_score_norm: 70, calculated_level: 'High', rule_risks: [], multipliers: [] } },
+      entity: {
+        name: 'mock-host',
+        type: 'node',
+        risk: { calculated_score_norm: 70, calculated_level: 'High' },
+      },
+      host: {
+        name: 'mock-host',
+        risk: {
+          calculated_score_norm: 70,
+          calculated_level: 'High',
+          rule_risks: [],
+          multipliers: [],
+        },
+      },
     };
 
     const userRecord = {
       '@timestamp': '2025-01-01T00:00:00.000Z',
       'user.name': 'mock-user',
-      entity: { name: 'mock-user', type: 'node', risk: { calculated_score_norm: 60, calculated_level: 'Medium' } },
-      user: { name: 'mock-user', risk: { calculated_score_norm: 60, calculated_level: 'Medium', rule_risks: [], multipliers: [] } },
+      entity: {
+        name: 'mock-user',
+        type: 'node',
+        risk: { calculated_score_norm: 60, calculated_level: 'Medium' },
+      },
+      user: {
+        name: 'mock-user',
+        risk: {
+          calculated_score_norm: 60,
+          calculated_level: 'Medium',
+          rule_risks: [],
+          multipliers: [],
+        },
+      },
     };
 
     const records = isHost ? [hostRecord] : isUser ? [userRecord] : [];

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -59,6 +59,62 @@ export const mockRiskEngineEnabled = () => {
   }).as('riskEngineStatus');
 };
 
+const ENTITY_STORE_ENTITIES_URL = '/api/security/entity_store/entities*';
+
+const entityStoreDsl = (entityIndex: string) =>
+  JSON.stringify({
+    index: [entityIndex],
+    body: { query: { bool: { filter: [{ exists: { field: 'entity.risk.calculated_score_norm' } }] } } },
+  });
+
+/**
+ * Intercepts the entity store v2 entities list API so that host/user risk score
+ * tables render with stub data in environments where the entity store index is
+ * not populated (e.g. CI inspect-button tests).
+ */
+export const mockEntityStoreRiskScores = () => {
+  const entityIndex = '.entities.v2.latest.security_default-00001';
+
+  cy.intercept('GET', ENTITY_STORE_ENTITIES_URL, (req) => {
+    const entityTypesParam = req.query?.entity_types;
+    const entityTypes: string = Array.isArray(entityTypesParam)
+      ? entityTypesParam.join(',')
+      : String(entityTypesParam ?? '');
+    const isHost = entityTypes.includes('host');
+    const isUser = entityTypes.includes('user');
+
+    const hostRecord = {
+      '@timestamp': '2025-01-01T00:00:00.000Z',
+      'host.name': 'mock-host',
+      entity: { name: 'mock-host', type: 'node', risk: { calculated_score_norm: 70, calculated_level: 'High' } },
+      host: { name: 'mock-host', risk: { calculated_score_norm: 70, calculated_level: 'High', rule_risks: [], multipliers: [] } },
+    };
+
+    const userRecord = {
+      '@timestamp': '2025-01-01T00:00:00.000Z',
+      'user.name': 'mock-user',
+      entity: { name: 'mock-user', type: 'node', risk: { calculated_score_norm: 60, calculated_level: 'Medium' } },
+      user: { name: 'mock-user', risk: { calculated_score_norm: 60, calculated_level: 'Medium', rule_risks: [], multipliers: [] } },
+    };
+
+    const records = isHost ? [hostRecord] : isUser ? [userRecord] : [];
+
+    req.reply({
+      statusCode: 200,
+      body: {
+        records,
+        page: 1,
+        per_page: 10,
+        total: records.length,
+        inspect: {
+          dsl: [entityStoreDsl(entityIndex)],
+          response: [JSON.stringify({ hits: { total: { value: records.length }, hits: [] } })],
+        },
+      },
+    });
+  }).as('entityStoreEntities');
+};
+
 export const openRiskInformationFlyout = () => cy.get(OPEN_RISK_INFORMATION_FLYOUT_BUTTON).click();
 
 export const openEntityStoreEnablementModal = () => {

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -94,6 +94,11 @@ const entityStoreDsl = (entityIndex: string) =>
  * Intercepts the entity store v2 entities list API so that host/user risk score
  * tables render with stub data in environments where the entity store index is
  * not populated (e.g. CI inspect-button tests).
+ *
+ * Coverage note: `inspect.dsl` always echoes the hardcoded `entityIndex`, so
+ * the table-case assertions on `INSPECT_MODAL_INDEX_PATTERN` are circular and
+ * cannot detect a wrong-index regression. The KPI lens-visualization cases
+ * (which use the search strategy, not this intercept) are unaffected.
  */
 export const mockEntityStoreRiskScores = () => {
   const entityIndex = '.entities.v2.latest.security_default-00001';

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -71,7 +71,9 @@ export const mockRiskEnginePrivileges = () => {
       privileges: {
         elasticsearch: {
           cluster: {},
-          index: {},
+          index: {
+            'risk-score.risk-score-*': { read: true, write: true },
+          },
         },
       },
     },

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/entity_analytics.ts
@@ -5,7 +5,10 @@
  * 2.0.
  */
 
-import { RISK_ENGINE_STATUS_URL } from '@kbn/security-solution-plugin/common/constants';
+import {
+  RISK_ENGINE_STATUS_URL,
+  RISK_ENGINE_PRIVILEGES_URL,
+} from '@kbn/security-solution-plugin/common/constants';
 import { BASIC_TABLE_LOADING } from '../screens/common';
 import {
   ANOMALIES_TABLE_ROWS,
@@ -50,13 +53,29 @@ export const navigateToNextPage = () => {
 };
 
 export const mockRiskEngineEnabled = () => {
-  // mock the risk engine status
   cy.intercept('GET', RISK_ENGINE_STATUS_URL, {
     statusCode: 200,
     body: {
       risk_engine_status: 'ENABLED',
     },
   }).as('riskEngineStatus');
+};
+
+export const mockRiskEnginePrivileges = () => {
+  cy.intercept('GET', RISK_ENGINE_PRIVILEGES_URL, {
+    statusCode: 200,
+    body: {
+      has_all_required: true,
+      has_read_permissions: true,
+      has_write_permissions: true,
+      privileges: {
+        elasticsearch: {
+          cluster: {},
+          index: {},
+        },
+      },
+    },
+  }).as('riskEnginePrivileges');
 };
 
 const ENTITY_STORE_ENTITIES_URL = '/api/security/entity_store/entities*';

